lru.go

```diff
@@ -38,10 +39,10 @@ func New(size int) (*Cache, error) {
 
 // Purge is used to completely clear the cache
 func (c *Cache) Purge() {
-	c.lock.Lock()
-	defer c.lock.Unlock()
-	c.evictList = list.New()
-	c.items = make(map[interface{}]*list.Element, c.size)
+	// Evict each element in the list.
```

Comment:
I did that initially and ran into a deadlock in the test. I'll take another look and see if I can do it with less locking.